File: valid_parentheses/valid_parentheses.py
def valid_parentheses(s):
    stack = []
    closed_brackets = {
        ')' : '(',
        '}' : '{',
        ']' : '['
    }
    for char in s:
        if char not in closed_brackets:
            stack.append(char)
        else:
            if not stack or stack.pop() != closed_brackets[char]:
                return False
    return not stack

s = '{[]}'
print("The string is a valid parenthesis: ", valid_parentheses(s))

File: valid_parentheses/valid_parentheses.ts
export {};

interface bracketsT {
  [key: string]: string;
}

function validParenthesis(s: string): boolean {
  const stack = [];
  const closed_brackets: bracketsT = {
    ")": "(",
    "}": "{",
    "]": "["
  };
  for (const char of s) {
    if (!(char in closed_brackets)) stack.push(char);
    else {
      if (stack.length === 0 || stack.pop() !== closed_brackets[char]) return false;
    }
  }
  return stack.length === 0;
}

const s = "{][]}";
console.log("The string is a valid parenthesis: ", validParenthesis(s));
